express/parsers/apps/aiida/formats/zip.py

```diff
@@ -0,0 +1,171 @@
+import json
+from zipfile import ZipFile
+from zipfile import BadZipFile
+
+import esse
+
+from express.parsers.apps.aiida.settings import SUPPORTED_AIIDA_ARCHIVE_VERSION
+from express.parsers.apps.aiida.settings import SUPPORTED_AIIDA_VERSION
+from express.parsers.apps.aiida.settings import TEMPLATES_MATERIALS_PATH
+
+
+ES = esse.ESSE()
+SCHEMA_MATERIAL = ES.get_schema_by_id('material')
+
+
+__all__ = [
+    'AiidaArchivefileParseError',
+    'AiidaZipParser',
+    'BadAiidaArchiveZipFile',
+    'BadZipFile',
+]
+
+
+class BadAiidaArchiveZipFile(ValueError):
+    """Error raised when parsing of archive file critically failed.
+
+    May indicate that the file may not actually be an
+    AiiDA archive file.
+    """
+
+    def __init__(self, path, error):
+        self.path = path
+        self.error = error
+
+
+class AiidaArchiveFileParseError(RuntimeError):
+    """Error raised when error occurred during parsing of archive file.
+    """
+
+    def __init__(self, path, reason):
+        self.path = path
+        self.reason = reason
+
+
+class AiidaZipParser:
+    """
+    Parser for AiiDA archive zip files.
+
+    Args:
+        zip_file_path: path to AiiDA archive file
+
+    Raises:
+        BadZipFile:
+            The file at the given path cannot be read as a valid zip file.
+        BadAiidaArchiveZipFile:
+            The file at the give path cannot be read as valid AiiDA archive file.
+    """
+
+    def __init__(self, zip_file_path):
+        self.zip_file_path = zip_file_path
+
+        self._metadata = None
+        self._data = None
+
+        self._check_supported_versions()  # triggers parsing of metadata
+
+    @property
+    def metadata(self):
+        if self._metadata is None:
+            with ZipFile(self.zip_file_path) as source:
+                self._metadata = json.loads(source.read('metadata.json'))
+        return self._metadata
+
+    @property
+    def data(self):
+        if self._data is None:
+            with ZipFile(self.zip_file_path) as source:
+                self._data = json.loads(source.read('data.json'))
+        return self._data
+
+    def _check_supported_versions(self):
+        """
+        Check whether the versions of the exported archive file are supported.
+
+        Args:
+            metadata (dict): deserialized metadata.json object
```

Comment:
I'm guessing metadata was converted to self.metadata at some point? This parameter doesn't exist anymore.